sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/ProcessorLiveTests.cs

```diff
@@ -309,7 +309,14 @@ Task ProcessErrors(ProcessErrorEventArgs args)
 
                 if (args.Exception is ServiceBusException sbException)
                 {
-                    if (sbException.Reason == ServiceBusException.FailureReason.MessagingEntityNotFound)
+                    if (sbException.Reason == ServiceBusException.FailureReason.MessagingEntityNotFound ||
+                        // There is a race condition wherein the service closes the connection when getting
+                        // the request for the non-existant queue. If the connection is closed by the time
+                        // our exception handling kicks in, we throw it as a ServiceCommunicationProblem
+                        // as we cannot be sure the error wasn't due to the connection being closed,
+                        // as opposed to what we know is the true cause in this case,
+                        // MessagingEntityNotFound.
+                        sbException.Reason == ServiceBusException.FailureReason.ServiceCommunicationProblem)
```

Comment:
Isn't `ServiceCommunicationProblem` a retryable error?

Reply:
Yes, it is

Reply:
Then why are we not retrying in this case?

Reply:
It would retry internally. This would be the exception from the final try.

Reply:
Ahh, got it.